src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2196,23 +2194,26 @@ public int GetLowerBound(int dimension)
 #if REAL_MULTIDIM_ARRAYS
             if (!IsSzArray)
             {
-                if ((dimension >= Rank) || (dimension < 0))
+                int rank = Rank;
+                if (dimension != 0 && (uint)dimension >= rank)
                     throw new IndexOutOfRangeException();
 
                 unsafe
                 {
                     fixed (int* pNumComponents = &_numComponents)
                     {
                         // Lower bounds follow after upper bounds.
-                        return *(pNumComponents + 1 + PADDING + Rank + dimension);
+                        return *(pNumComponents + 1 + PADDING + rank + dimension);
                     }
                 }
             }
 #else
             MDArray mdArray = this as MDArray;
             if (mdArray != null)
             {
-                if ((dimension >= mdArray.MDRank) || (dimension < 0))
+                // If the dimension is 0 or 1, we don't have to call get_MDRank
+                // which is a virtual method call
+                if ((dimension & ~1) == 0 && (uint)dimension >= mdArray.MDRank)
```

Comment:
We do not use MDArray.cs for CoreRT so there is no good way to test it, and we are on path to eliminate it completely - could you please revert the MDArray related changes?